src/en/toonily/src/eu/kanade/tachiyomi/extension/en/toonily/Toonily.kt

```diff
@@ -17,63 +17,71 @@ class Toonily : Madara(
     "en",
     SimpleDateFormat("MMM d, yy", Locale.US),
 ) {
-
-    override val client: OkHttpClient = super.client.newBuilder()
+    override val client = super.client.newBuilder()
         .addNetworkInterceptor(CookieInterceptor(domain, "toonily-mature" to "1"))
+        .addInterceptor(::hdCoverInterceptor)
         .build()
 
-    override val mangaSubString = "webtoon"
-
-    private fun searchPage(page: Int, query: String): String {
-        val urlQuery = query.trim()
-            .lowercase(Locale.US)
-            .replace(titleSpecialCharactersRegex, "-")
-            .replace(trailingHyphenRegex, "")
-            .let { if (it.isNotEmpty()) "$it/" else it }
-        return if (page > 1) {
-            "search/${urlQuery}page/$page/"
-        } else {
-            "search/$urlQuery"
-        }
-    }
-
-    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        val request = super.searchMangaRequest(page, query, filters)
-
-        val queries = request.url.queryParameterNames
-            .filterNot { it == "s" }
-
-        val newUrl = "$baseUrl/${searchPage(page, query)}".toHttpUrl().newBuilder().apply {
-            queries.map { q ->
-                request.url.queryParameterValues(q).map {
-                    this.addQueryParameter(q, it)
-                }
-            }
-        }.build()
+    override val mangaSubString = "serie"
+    override val filterNonMangaItems = false
+    override val useNewChapterEndpoint = true
+    override val sendViewCount = false
+    override val useLoadMoreRequest = LoadMoreStrategy.Always
 
-        return request.newBuilder()
-            .url(newUrl)
-            .build()
-    }
+    override fun searchMangaSelector() = "div.page-item-detail.manga"
 
     override fun genresRequest(): Request {
         return GET("$baseUrl/search/?post_type=wp-manga", headers)
     }
 
-    // The source customized the Madara theme and broke the filter.
-    override val filterNonMangaItems = false
-
-    override val useNewChapterEndpoint: Boolean = true
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val newManga = SManga.create().apply {
+            url = manga.url.replace("/webtoon/", "/$mangaSubString/")
```

Comment:
I guess this is to avoid migration? You will run into the other issue where browsing will show library items as non-library items. Usually I think the tradeoff here is worth it.

If you want to address it further you would do the opposite- always use the old structure and rewrite it in the interceptor. Downside here is that you are stuck with the old URL format forever.

Reply:
this is just meant to prevent redirects